internal/cloudapi/v2/compose_test.go

```diff
@@ -87,6 +87,73 @@ func TestGetBlueprintWithCustomizations(t *testing.T) {
 				Enabled:  common.ToPtr(true),
 			},
 		},
+		Firewall: &FirewallCustomization{
+			Ports: common.ToPtr([]string{
+				"22/tcp",
+			}),
+		},
+		Hostname:           common.ToPtr("hostname"),
+		InstallationDevice: common.ToPtr("/dev/sda"),
+		Kernel: &struct {
```

Comment:
Why embed structs here instead of using the types from the generator?

Reply:
In that case I'd have to split the sub-objects out in the spec to be a separate entry under schemas, objects which are immediately defined are also nested in the generated code. 

https://github.com/croissanne/osbuild-composer/commit/9cbf8b26d8504aae578b46400aa8d23e2b104d1f#diff-8bbd501fd1c37ce113701845557026187ef15239c63c0a284961a72517f6891bR373-R380

Reply:
Oh, I see. This more of the openapi generator not knowing how to make decent Go code :/ But if you have a way to make it behave I'd pick that over this embedded struct stuff. We ought to be able to have properly defined structs in one place instead of needing to copy-paste these embedded things everywhere.

Reply:
will do :)